File: backend/src/main/java/z9/hobby/domain/checkin/service/CheckInService.java
package z9.hobby.domain.checkin.service;

import java.util.List;
import z9.hobby.domain.checkin.dto.CheckInRequestDto;
import z9.hobby.domain.checkin.dto.CheckInResponseDto;


public interface CheckInService {
    void createCheckIn(Long userId, CheckInRequestDto.CheckInDto requestDto);
    void updateCheckIn(Long userId, CheckInRequestDto.CheckInDto requestDto);
    List<CheckInResponseDto.ResponseData> getAllCheckIns(Long scheduleId);
    CheckInResponseDto.ResponseData getMyCheckIn(Long scheduleId, Long userId);
}

File: backend/src/main/java/z9/hobby/domain/checkin/service/CheckInService.kt
package z9.hobby.domain.checkin.service

import z9.hobby.domain.checkin.dto.CheckInRequestDto
import z9.hobby.domain.checkin.dto.CheckInResponseDto

interface CheckInService {
    fun createCheckIn(userId: Long, requestDto: CheckInRequestDto)
    fun updateCheckIn(userId: Long, requestDto: CheckInRequestDto)
    fun checkInProcess(userId: Long, requestDto: CheckInRequestDto)
    fun getAllCheckIns(scheduleId: Long): List<CheckInResponseDto>
    fun getMyCheckIn(scheduleId: Long, userId: Long): CheckInResponseDto
}
